src/Module.zig

```diff
@@ -2128,7 +2133,10 @@ pub const File = struct {
         file.multi_pkg = true;
         file.status = .astgen_failure;
 
-        std.debug.assert(file.zir_loaded);
+        // We can only mark children as failed if the ZIR is loaded, which may not
+        // be the case if there were other astgen failures in this file
+        if (!file.zir_loaded) return;
```

Comment:
What happened to my beautiful assertion? We can't rely on the ZIR being loaded anymore?

Reply:
See the comment and #14499. Basically, it's possible to get a multi-package error on a file with other AstGen failures, so there are valid cases when the ZIR won't be loaded: in these cases, we've not triggered analysis for any dependencies, so there's nothing to mark as `multi_pkg`.